### PR TITLE
Consolidate duplicated clang-format files

### PR DESCRIPTION
**Pull request by NickGerleman**
This consolidates three near-identical clang-format files that were in the root of the repo, packages directory, and vnext directory. The new config file is based off of vnext config file.

This doesn't lead to any formatting changes, since the pacakges + vnext clang-format files both already had overriden 120 width line length.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7021)

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -35,15 +35,15 @@ BreakBeforeTernaryOperators: true
...
-ColumnLimit:       80
+ColumnLimit:       120
 CommentPragmas:  '^ IWYU pragma:'
 ConstructorInitializerAllOnOneLineOrOnePerLine: true
 ConstructorInitializerIndentWidth: 4
 ContinuationIndentWidth: 4
 Cpp11BracedListStyle: true
 DerivePointerAlignment: false
 DisableFormat:   false
-ForEachMacros:   [ FOR_EACH_RANGE, FOR_EACH, TEST_METHOD ]
```
Comment (NickGerleman) [inline .clang-format:46]: vnext and packages configs did not have TEST_METHOD. Results were much better following the omission as in vnext.

File: package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "change": "beachball change",
     "clean": "lage clean -- --color",
     "format": "format-files -i -style=file -assume-filename=../.clang-format",
-    "format:verify": "format-files -i -style=file -assume-filename=../.clang-format -verify",
```
Comment (NickGerleman) [inline package.json:12]: `-assume-filename=../.clang-format` was being used incorrectly. It's used to override language auto-detection.

> -assume-filename= option can be used to override the file name clang-format uses to detect the language.

File: .clang-format [hunk 3]
```diff
@@ -35,15 +35,15 @@ BreakBeforeTernaryOperators: true
 BreakConstructorInitializersBeforeComma: false
 BreakAfterJavaFieldAnnotations: false
 BreakStringLiterals: false
-ColumnLimit:       80
+ColumnLimit:       120
```
Comment (dannyvv) [inline .clang-format:38]: Thank you!

**Review by dannyvv — APPROVED** [review]

**Comment by NickGerleman** [comment]
@msftbot merge in 20 hours

**Pull request merged**